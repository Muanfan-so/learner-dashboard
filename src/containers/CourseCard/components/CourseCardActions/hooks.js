import { handleEvent } from 'data/services/segment/utils';
import { eventNames } from 'data/services/segment/constants';

export const useCourseCardActionData = () => {
  const trackUpgradeClick = () => {
    handleEvent(eventNames.upgradeCourse, {
      pageName: 'learner_home',
      linkType: 'button',
      linkCategory: 'green_upgrade',
    });
  };

  return {
    trackUpgradeClick,
  };
};

export default useCourseCardActionData;
