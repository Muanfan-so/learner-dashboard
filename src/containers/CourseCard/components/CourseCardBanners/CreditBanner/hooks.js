import { StrictDict } from 'utils';
import { hooks as appHooks } from 'data/redux';

import ApprovedContent from './views/ApprovedContent';
import EligibleContent from './views/EligibleContent';
import MustRequestContent from './views/MustRequestContent';
import PendingContent from './views/PendingContent';
import RejectedContent from './views/RejectedContent';

export const statusComponents = StrictDict({
  pending: PendingContent,
  approved: ApprovedContent,
  rejected: RejectedContent,
});

export const useCreditBannerData = (cardId) => {
  const credit = appHooks.useCardCreditData(cardId);
  const { supportEmail } = appHooks.usePlatformSettingsData();
  if (!credit.isEligible) { return null; }

  const { error, purchased, requestStatus } = credit;
  let ContentComponent = EligibleContent;
  if (purchased) {
    if (requestStatus == null) {
      ContentComponent = MustRequestContent;
    } else if (Object.keys(statusComponents).includes(requestStatus)) {
      ContentComponent = statusComponents[requestStatus];
    }
    // Current behavior is to show Elligible State if unknown request status is returned
  }
  return {
    ContentComponent,
    error,
    supportEmail,
  };
};

export default {
  useCreditBannerData,
};
