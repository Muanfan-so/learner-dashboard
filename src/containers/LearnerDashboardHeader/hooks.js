import React from 'react';
import { useWindowSize, breakpoints } from '@edx/paragon';
import track from 'tracking';

export const useIsCollapsed = () => {
  const { width } = useWindowSize();
  const isCollapsed = React.useMemo(() => (width <= breakpoints.large.maxWidth), [width]);
  return isCollapsed;
};

export const findCoursesNavClicked = (href) => track.findCourses.findCoursesClicked(href, {
  linkName: 'learner_home_nav_discover',
});

export const findCoursesNavDropdownClicked = (href) => track.findCourses.findCoursesClicked(href, {
  linkName: 'learner_home_nav_dropdown_discover',
});

export default {
  useIsCollapsed,
  findCoursesNavClicked,
  findCoursesNavDropdownClicked,
};
