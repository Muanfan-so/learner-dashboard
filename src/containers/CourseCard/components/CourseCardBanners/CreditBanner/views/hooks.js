import React from 'react';

import { StrictDict } from 'utils';
import { apiHooks } from 'hooks';

import * as module from './hooks';

export const state = StrictDict({
  creditRequestData: (val) => React.useState(val), // eslint-disable-line
});

export const useCreditRequestData = (cardId) => {
  const [requestData, setRequestData] = module.state.creditRequestData(null);
  const createCreditApiRequest = apiHooks.useCreateCreditRequest(cardId);
  const createCreditRequest = (e) => {
    e.preventDefault();
    createCreditApiRequest().then(setRequestData);
  };
  return { requestData, createCreditRequest };
};

export default {
  useCreditRequestData,
};
