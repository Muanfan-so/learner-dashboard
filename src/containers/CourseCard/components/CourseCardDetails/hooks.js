import { useIntl } from '@edx/frontend-platform/i18n';
import { hooks as appHooks } from 'data/redux';

import * as hooks from './hooks';
import messages from './messages';

export const useAccessMessage = ({ cardId }) => {
  const { formatMessage, formatDate } = useIntl();
  const enrollment = appHooks.useCardEnrollmentData(cardId);
  const courseRun = appHooks.useCardCourseRunData(cardId);
  if (!courseRun.isStarted) {
    if (!courseRun.startDate) { return null; }
    const startDate = formatDate(courseRun.startDate);
    return formatMessage(messages.courseStarts, { startDate });
  }
  if (enrollment.isEnrolled) {
    const { isArchived, endDate } = courseRun;
    const {
      accessExpirationDate,
      isAuditAccessExpired,
    } = enrollment;

    if (enrollment.isAudit && accessExpirationDate) {
      return formatMessage(
        isAuditAccessExpired ? messages.accessExpired : messages.accessExpires,
        { accessExpirationDate: formatDate(accessExpirationDate) },
      );
    }
    if (!endDate) { return null; }
    return formatMessage(
      isArchived ? messages.courseEnded : messages.courseEnds,
      { endDate: formatDate(endDate) },
    );
  }
  return null;
};

export const useCardDetailsData = ({ dispatch, cardId }) => {
  const { formatMessage } = useIntl();
  const providerName = appHooks.useCardProviderData(cardId).name;
  const { courseNumber } = appHooks.useCardCourseData(cardId);
  const {
    isEntitlement,
    isFulfilled,
    canChange,
  } = appHooks.useCardEntitlementData(cardId);

  const openSessionModal = appHooks.useUpdateSelectSessionModalCallback(dispatch, cardId);

  return {
    providerName: providerName || formatMessage(messages.unknownProviderName),
    accessMessage: hooks.useAccessMessage({ cardId }),
    isEntitlement,
    isFulfilled,
    canChange,
    openSessionModal,
    courseNumber,
    changeOrLeaveSessionMessage: formatMessage(messages.changeOrLeaveSessionButton),
  };
};

export default useCardDetailsData;
