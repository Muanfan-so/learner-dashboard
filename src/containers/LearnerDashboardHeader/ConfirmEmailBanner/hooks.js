import React from 'react';

import { StrictDict } from 'utils';
import { hooks as appHooks, thunkActions } from 'data/redux';

import { useDispatch } from 'react-redux';

import * as module from './hooks';

export const state = StrictDict({
  showPageBanner: (val) => React.useState(val), // eslint-disable-line
  showConfirmModal: (val) => React.useState(val), // eslint-disable-line
});

export const useConfirmEmailBannerData = () => {
  const dispatch = useDispatch();
  const { isNeeded } = appHooks.useEmailConfirmationData();
  const [showPageBanner, setShowPageBanner] = module.state.showPageBanner(isNeeded);
  const [showConfirmModal, setShowConfirmModal] = module.state.showConfirmModal(false);
  const closePageBanner = () => setShowPageBanner(false);
  const closeConfirmModal = () => setShowConfirmModal(false);
  const openConfirmModal = () => setShowConfirmModal(true);

  const openConfirmModalButtonClick = () => {
    dispatch(thunkActions.app.sendConfirmEmail());
    openConfirmModal();
  };

  const userConfirmEmailButtonClick = () => {
    closeConfirmModal();
    closePageBanner();
  };
  return {
    isNeeded,
    showPageBanner,
    closePageBanner,
    showConfirmModal,
    closeConfirmModal,
    openConfirmModalButtonClick,
    userConfirmEmailButtonClick,
  };
};

export default useConfirmEmailBannerData;
