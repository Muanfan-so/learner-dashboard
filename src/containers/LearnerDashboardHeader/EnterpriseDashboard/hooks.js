import React from 'react';
import { hooks as appHooks } from 'data/redux';
import { StrictDict } from 'utils';
import * as module from './hooks';

export const state = StrictDict({
  showDialog: (val) => React.useState(val), // eslint-disable-line
  selectedItem: (val) => React.useState(val), // eslint-disable-line
});

export const useEnterpriseDashboardHook = () => {
  const { availableDashboards, mostRecentDashboard } = appHooks.useEnterpriseDashboardData();
  const [showDialog, setShowDialog] = module.state.showDialog(false);
  const [selectedItem, setSelectedItem] = module.state.selectedItem({});

  const beginSelectDashboardItem = (val) => () => {
    setSelectedItem(val);
    setShowDialog(true);
  };

  const cancelSelectDashboardItem = () => {
    setSelectedItem({});
    setShowDialog(false);
  };

  return {
    availableDashboards,
    mostRecentDashboard,
    showDialog,

    selectedItem,
    beginSelectDashboardItem,
    cancelSelectDashboardItem,
  };
};

export default useEnterpriseDashboardHook;
