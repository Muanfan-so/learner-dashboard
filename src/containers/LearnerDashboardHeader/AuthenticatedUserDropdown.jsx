import React from 'react';
import PropTypes from 'prop-types';

import { getConfig } from '@edx/frontend-platform';
import { useIntl } from '@edx/frontend-platform/i18n';
import { Dropdown, Icon } from '@edx/paragon';
import { Person } from '@edx/paragon/icons';

import messages from './messages';
import EnterpriseDashboard from './EnterpriseDashboard';

export const AuthenticatedUserDropdown = ({ username }) => {
  const { formatMessage } = useIntl();
  return (
    <>
      <Dropdown className="user-dropdown">
        <Dropdown.Toggle id="user" variant="primary">
          <Icon src={Person} />
          <span data-hj-suppress className="d-none d-md-inline">
            {username}
          </span>
        </Dropdown.Toggle>
        <Dropdown.Menu className="dropdown-menu-right">
          <EnterpriseDashboard />
          <Dropdown.Item href={`${getConfig().LMS_BASE_URL}/u/${username}`}>
            {formatMessage(messages.profile)}
          </Dropdown.Item>
          <Dropdown.Item href={`${getConfig().LMS_BASE_URL}/account/settings`}>
            {formatMessage(messages.account)}
          </Dropdown.Item>
          {getConfig().ORDER_HISTORY_URL && (
            <Dropdown.Item href={getConfig().ORDER_HISTORY_URL}>
              {formatMessage(messages.orderHistory)}
            </Dropdown.Item>
          )}
          <Dropdown.Item href={getConfig().SUPPORT_URL}>
            {formatMessage(messages.help)}
          </Dropdown.Item>
          <Dropdown.Item href={getConfig().LOGOUT_URL}>
            {formatMessage(messages.signOut)}
          </Dropdown.Item>
        </Dropdown.Menu>
      </Dropdown>
    </>
  );
};

AuthenticatedUserDropdown.propTypes = {
  username: PropTypes.string.isRequired,
};

export default AuthenticatedUserDropdown;
