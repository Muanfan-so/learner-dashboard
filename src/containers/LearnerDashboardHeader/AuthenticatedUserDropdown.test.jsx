import { shallow } from 'enzyme';

import { reduxHooks } from 'hooks';
import { AuthenticatedUserDropdown } from './AuthenticatedUserDropdown';
import { useIsCollapsed } from './hooks';

jest.mock('@edx/frontend-platform/react', () => ({
  AppContext: {
    authenticatedUser: {
      profileImage: 'profileImage',
    },
  },
}));
jest.mock('hooks', () => ({
  reduxHooks: {
    useEnterpriseDashboardData: jest.fn(),
    usePlatformSettingsData: jest.fn(() => ({
      courseSearchUrl: 'test-course-search-url',
    })),
  },
}));
jest.mock('containers/LearnerDashboardHeader/hooks', () => ({
  useIsCollapsed: jest.fn(),
  findCoursesNavDropdownClicked: (href) => jest.fn().mockName(`findCoursesNavDropdownClicked('${href}')`),
}));

describe('AuthenticatedUserDropdown', () => {
  const props = {
    username: 'username',
  };
  const defaultDashboardData = {
    label: 'label',
    url: 'url',
  };

  describe('snapshots', () => {
    test('with enterprise dashboard', () => {
      reduxHooks.useEnterpriseDashboardData.mockReturnValueOnce(defaultDashboardData);
      useIsCollapsed.mockReturnValueOnce(true);
      const wrapper = shallow(<AuthenticatedUserDropdown {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
    test('without enterprise dashboard and expanded', () => {
      reduxHooks.useEnterpriseDashboardData.mockReturnValueOnce(null);
      useIsCollapsed.mockReturnValueOnce(false);
      const wrapper = shallow(<AuthenticatedUserDropdown {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
