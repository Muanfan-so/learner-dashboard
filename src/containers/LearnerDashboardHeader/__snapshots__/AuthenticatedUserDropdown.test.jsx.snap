// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticatedUserDropdown snapshots with enterprise dashboard 1`] = `
<Dropdown
  className="user-dropdown"
  variant="light"
>
  <Dropdown.Toggle
    id="user"
    src="profileImage"
    variant="primary"
  >
    <span
      className="d-none d-md-inline"
      data-hj-suppress={true}
    >
      username
    </span>
  </Dropdown.Toggle>
  <Dropdown.Menu
    className="dropdown-menu-right"
  >
    <Dropdown.Header>
      SWITCH DASHBOARD
    </Dropdown.Header>
    <Dropdown.Item
      as="a"
      className="active"
      href="/edx-dashboard"
    >
      Personal
    </Dropdown.Item>
    <Dropdown.Item
      as="a"
      href="url"
      key="label"
    >
      label
       
      Dashboard
    </Dropdown.Item>
    <Dropdown.Divider />
    <Dropdown.Item
      href="http://localhost:18000/u/username"
    >
      Profile
    </Dropdown.Item>
    <Dropdown.Item
      href="http://localhost:18000/dashboard/programs"
    >
      View Programs
    </Dropdown.Item>
    <Dropdown.Item
      href="test-course-search-url"
      onClick={[MockFunction findCoursesNavDropdownClickedtest-course-search-url]}
    >
      Explore courses
    </Dropdown.Item>
    <Dropdown.Item
      href="http://localhost:18000/account/settings"
    >
      Account
    </Dropdown.Item>
    <Dropdown.Item
      href="http://localhost:18000/support"
    >
      Help
    </Dropdown.Item>
    <Dropdown.Divider />
    <Dropdown.Item
      href="http://localhost:18000/logout"
    >
      Sign Out
    </Dropdown.Item>
  </Dropdown.Menu>
</Dropdown>
`;

exports[`AuthenticatedUserDropdown snapshots without enterprise dashboard and expanded 1`] = `
<Dropdown
  className="user-dropdown"
  variant="dark"
>
  <Dropdown.Toggle
    id="user"
    src="profileImage"
    variant="primary"
  >
    <span
      className="d-none d-md-inline"
      data-hj-suppress={true}
    >
      username
    </span>
  </Dropdown.Toggle>
  <Dropdown.Menu
    className="dropdown-menu-right"
  >
    <Dropdown.Header>
      SWITCH DASHBOARD
    </Dropdown.Header>
    <Dropdown.Item
      as="a"
      className="active"
      href="/edx-dashboard"
    >
      Personal
    </Dropdown.Item>
    <Dropdown.Divider />
    <Dropdown.Item
      href="http://localhost:18000/u/username"
    >
      Profile
    </Dropdown.Item>
    <Dropdown.Item
      href="http://localhost:18000/account/settings"
    >
      Account
    </Dropdown.Item>
    <Dropdown.Item
      href="http://localhost:18000/support"
    >
      Help
    </Dropdown.Item>
    <Dropdown.Divider />
    <Dropdown.Item
      href="http://localhost:18000/logout"
    >
      Sign Out
    </Dropdown.Item>
  </Dropdown.Menu>
</Dropdown>
`;
